com.unity.ml-agents/Runtime/Policies/BrainParameters.cs

```diff
@@ -50,8 +51,36 @@ public class BrainParameters
         [FormerlySerializedAs("numStackedVectorObservations")]
         [Range(1, 50)] public int NumStackedVectorObservations = 1;
 
+        [SerializeField]
+        internal int m_NumContinuousActions = 0;
+        [SerializeField]
+        internal int[] m_DiscreteBranchSizes = null;
+
+        /// <summary>
+        /// The specification of the Action space for the BrainParameters.
+        /// </summary>
+        public ActionSpec ActionSpec
+        {
+            get { return new ActionSpec(m_NumContinuousActions, m_DiscreteBranchSizes); }
```

Comment:
I think this means that someone could do something like
```csharp
myBrainParams.ActionSpec.NumContinuousActions = 4;
```
which would compile are run, but it would set NumContinuousActions on the new ActionSpec that gets returned here, but it wouldn't propagate to m_NumContinuousActions.

@surfnerd is there a better pattern for this?

Reply:
ActionSpec is a value type, so 'new' is redundant here I believe.

Reply:
without 'new' there will be compile error

Reply:
Now that the deprecated fields can be synced with corresponding new field, it's actually easier to just bring back ActionSpec and sync with ActionSpec fields to solve this problem.